Update README with rebranding message (#414)
AMDResearch/omnitrace: README.md

[![Documentation](https://github.com/ROCm/omnitrace/actions/workflows/docs.yml/badge.svg)](https://github.com/ROCm/omnitrace/actions/workflows/docs.yml)

> [!NOTE]
> Omnitrace is being rebranded to ROCm Systems Profiler and its new home is <https://github.com/ROCm/rocprofiler-systems>.
All future development will occur in the new repository; this includes upgrading the tool to use [rocprofiler-sdk](https://github.com/ROCm/rocprofiler-sdk).

> This repository will remain open for some time and can be used with versions of ROCm before the introduction of rocprofiler-sdk (ie., pre ROCm-6.2).

Reviewer: ```suggestion
This repository will remain open for some time and can be used with versions of ROCm before the introduction of rocprofiler-sdk (that is, before ROCm version 6.2).
```
